crates/holochain/src/core/workflow/sys_validation_workflow/query.rs

```diff
@@ -0,0 +1,94 @@
+use holo_hash::DhtOpHash;
+use holochain_sqlite::db::ReadManager;
+use holochain_sqlite::db::WriteManager;
+use holochain_state::query::prelude::*;
+use holochain_types::dht_op::DhtOp;
+use holochain_types::dht_op::DhtOpType;
+use holochain_types::env::EnvRead;
+use holochain_types::env::EnvWrite;
+use holochain_zome_types::Entry;
+use holochain_zome_types::SignedHeader;
+use holochain_zome_types::ValidationStatus;
+use rusqlite::named_params;
+
+use crate::core::workflow::error::WorkflowResult;
+
+/// Get all ops that need to sys validated in order.
+/// - Pending or awaiting sys dependencies.
+/// - Ordered by type then timestamp (See [`DhtOpOrder`])
+pub fn get_ops_to_sys_validate(env: EnvRead) -> WorkflowResult<Vec<DhtOp>> {
+    let results = env.conn()?.with_reader(|txn| {
+        let mut stmt = txn.prepare(
+            "
+            SELECT 
+            Header.blob as header_blob,
+            Entry.blob as entry_blob,
+            DhtOp.type as dht_type
+            JOIN
+            DhtOp ON DhtOp.header_hash = Header.hash
+            LEFT JOIN
+            Entry ON (Header.entry_hash IS NULL OR Header.entry_hash = Entry.hash)
+            WHERE
+            (DhtOp.validation_status IS NULL OR DhtOp.validation_stage = 1)
+            ORDER BY 
+            CASE DhtOp.type
+              WHEN :activity                THEN 0
+              WHEN :store_entry             THEN 1
+              WHEN :store_element           THEN 2
+              WHEN :updated_content         THEN 3
+              WHEN :updated_element         THEN 4
+              WHEN :deleted_by              THEN 5
+              WHEN :deleted_entry_header    THEN 6
+              WHEN :add_link                THEN 7
+              WHEN :remove_link             THEN 8
+            END,
```

Comment:
We might want to benchmark this in the future, it might be faster to let the type field be indexed as an INT where we choose the int values to produce this ordering.